Source/WebCore/platform/SharedBuffer.h

```diff
@@ -357,7 +340,10 @@ class SharedBufferBuilder {
 
     template <typename... Args>
     SharedBufferBuilder(std::in_place_t, Args&&... args)
-        : m_buffer(FragmentedSharedBuffer::create(std::forward<Args>(args)...)) { }
+        : m_buffer(SharedBuffer::create(std::forward<Args>(args)...))
+    {
+        m_buffer->m_contiguous = m_buffer->segmentsCount() <= 1;
```

Comment:
It would be nice if `SharedBuffer::create` would set its m_contiguous value itself.